thinc/layers/array_getitem.py

```diff
@@ -1,9 +1,14 @@
+import sys
 from typing import Union, Sequence, Tuple, TypeVar
 from ..types import ArrayXd, FloatsXd, IntsXd
 from ..model import Model
 
+if sys.version_info < (3, 8):
+    from typing_extensions import SupportsIndex
+else:
+    from typing import SupportsIndex
```

Comment:
We can move this bit to `thinc.compat` and then import `SupportsIndex` from there.

Reply:
Oh sorry, actually this should go into `thinc.types` instead!

Reply:
Done!

Sicne the intention is to make `SupportsIndex` available as an public symbol in `thinc.types`, I've opted to do what PEP 484 wants you to do for stub files, which is to re-export the imported symbol as the same name. Pyright/Pylance _really_ doesn't like it if you don't do this (or explicitly declare a `__all__` full of literal strings) although mypy tends to be okay. In either case, it doesn't hurt anything.